### PR TITLE
Add: FPC 0.5mm pitch top and bottom connector JLC parts numbers

### PR DESCRIPTION
**Pull request by Suke0811**
This should cover most of the popular P0.5 fpc sizes. FCC and FCA are just different packages.

### CODE REVIEW

**Comment by Suke0811** [comment]
Change the footprint to TE connectors that are closer to the AF07 in the future once tested.

File: edg/parts/PassiveConnector_Fpc.py [hunk 1]
```diff
@@ -75,14 +102,47 @@ def part_footprint_mfr_name(self, length: int) -> Tuple[str, str, str]:
 class Afc07Top(Fpc050Top, FootprintPassiveConnector, JlcPart):
   """Jushuo AFC07 series slide-lock top-contact 0.5mm-pitch FPC connectors, with partial JLC numbers for some parts
   and re-using the probably-compatible but not-purpose-designed FH12 footprint."""
-  _afc07_pins = set(range(4, 60+1))  # as listed by the part table
+  _afc07_pins = set(range(4, 60 + 1))  # as listed by the part table
   allowed_pins = _afc07_pins.intersection(HiroseFh12sh._kicad_pins)
   PART_NUMBERS = {  # partial list of the ones currently used
+    4: 'C2764271',
+    5: 'C262230',
+    # 5: 'C262578',
```
Comment (ducky64) [inline edg/parts/PassiveConnector_Fpc.py:110]:
```suggestion
    5: 'C262230',  # also C262578 in different packaging
```
maybe condense into one line (and below instances)

**Review by ducky64 — CHANGES_REQUESTED** [review]
Style improvement suggestions.

Can you also make sure both part number format strings include an asterisk for the packaging,?

**Comment by ducky64** [comment]
Also we can change the footprint in a separate PR. I'm planning to run a separate test panel that includes the footprint test.

File: edg/parts/PassiveConnector_Fpc.py [hunk 2]
```diff
@@ -54,16 +54,43 @@ def part_footprint_mfr_name(self, length: int) -> Tuple[str, str, str]:
...
+    45: 'C13507',  # FCC
     50: 'C262676',
+    # 50:'C262284', # FCC
+    54: 'C262677',
+    60: 'C2918970'  # FCC
   }
+
   def part_footprint_mfr_name(self, length: int) -> Tuple[str, str, str]:
     # TODO this isn't the intended hook and uses side effects, but it works for now
     self.assign(self.lcsc_part, self.PART_NUMBERS[length])
```
[the remark this answers is not in the document]
Comment (Suke0811) [reply edg/parts/PassiveConnector_Fpc.py:96]:
```
f"AFC01-S{length:02d}FC*-00"
f"AFC07-S{length:02d}EC*-00"
```

File: edg/parts/PassiveConnector_Fpc.py [hunk 3]
```diff
@@ -75,14 +102,47 @@ def part_footprint_mfr_name(self, length: int) -> Tuple[str, str, str]:
...
   and re-using the probably-compatible but not-purpose-designed FH12 footprint."""
-  _afc07_pins = set(range(4, 60+1))  # as listed by the part table
+  _afc07_pins = set(range(4, 60 + 1))  # as listed by the part table
   allowed_pins = _afc07_pins.intersection(HiroseFh12sh._kicad_pins)
   PART_NUMBERS = {  # partial list of the ones currently used
+    4: 'C2764271',
+    5: 'C262230',
+    # 5: 'C262578',
+    6: 'C413943',
+    7: '',  # 46
```
[the remark this answers is not in the document]
Comment (Suke0811) [reply edg/parts/PassiveConnector_Fpc.py:112]: Added the missing 7 pos parts number

**Comment by Suke0811** [comment]
How's this change[edae225](https://github.com/BerkeleyHCI/PolymorphicBlocks/pull/378/commits/edae22598e4da5acb22e49088752961e9b68043c)?

**Review by ducky64 — APPROVED** [review]

**Pull request merged**